boefjes/boefjes/plugins/kat_nmap_tcp/main.py

```diff
@@ -18,4 +18,11 @@ def run(boefje_meta: dict):
         cmd.append("-6")
 
     cmd.extend(["-oX", "-", str(ip)])
-    return [({"openkat/nmap-output"}, subprocess.run(cmd, capture_output=True).stdout.decode())]
+
+    output = subprocess.run(cmd, capture_output=True)
+    results = [({"openkat/nmap-output"}, output.stdout.decode())]
+
+    if output.stderr:
+        return results.append(({"error/boefje"}, output.stderr.decode()))
```

Comment:
```suggestion
        if not output.stdout:
            results = []
        return results.append(({"error/boefje"}, output.stderr.decode()))
```

Reply:
I also made a push. That should also cover this, right?

Reply:
Your solution makes sure we don't Delete any previous proofs. 
My suggestions allows for a situation where we still have some logical output that we'd want to ingest into the graph, but also some errors (which did not stop us from creating data). The million dollar question is: 'is there a situation where nmap throws errors, but still manages to produce valid output?'

Reply:
Haha yeah good question. To be honest I don't think so. Im more inclined to be on the safe side and never delete old observations. If you feel like there are situations when nmap throws errors & good output, then I'm happy to rewrite it to your solution.